From: Jason Rhinelander <jason@imaginary.ca>
Date: Thu, 25 Jul 2019 18:32:13 -0300
Subject: 'make check' runs test suite

debhelper prefers a make test target, but that target is heavily
restricted in cmake; this removes it and adds a check target that works
instead to run the test suite in the build.
---
 CMakeLists.txt      | 2 +-
 test/CMakeLists.txt | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c411c8..a51f123 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -314,7 +314,7 @@ else()
   endif(ANDROID)
 endif(SHADOW)
 
-enable_testing()
+#enable_testing()
 
 if (NOT SHADOW)
   add_subdirectory(test)
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index c7b5e84..95ee0fe 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -74,3 +74,6 @@ endif(NOT WIN32)
 if(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
   target_link_directories(${TEST_EXE} PRIVATE /usr/local/lib)
 endif()
+
+add_custom_target(check COMMAND ${TEST_EXE})
+
